src/lib/utils/errors.ts

```diff
@@ -0,0 +1,70 @@
+import { captureException } from '@sentry/browser';
+import { ScopeContext } from '@sentry/types/types/scope';
+
+type BalancerExceptionAction = 'swap' | 'joinPool' | 'exitPool';
+
+export function captureBalancerException(
+  error: Error,
+  action: BalancerExceptionAction,
+  messagePrefix: string,
+  captureContext?: Partial<ScopeContext>
+): void {
+  const { reason, balError } = getReasonAndBalErrorFromError(error);
+  const tags: { [key: string]: string } = { ...captureContext?.tags, action };
+
+  if (balError) {
+    tags.balError = balError;
+  }
+
+  captureException(
+    getErrorForAction(action, `${messagePrefix}: ${reason}`, error),
+    {
+      ...captureContext,
+      extra: {
+        ...captureContext?.extra,
+        reason,
+        balError,
+      },
+      tags,
+    }
+  );
+}
+
+function getReasonAndBalErrorFromError(error: Error): {
+  reason: string;
+  balError: string | null;
+} {
+  const reason: string =
+    (error as { reason?: string }).reason || 'no reason available';
+  const balError = reason.match(/BAL#([0-9])\d\d/g);
```

Comment:
Why are you using `[0-9]` for the first number and `\d` for the others? This could be simplified to `/BAL#[0-9]{3}/g`

Reply:
updated 👍